tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -292,6 +322,72 @@ var AppInstallManager = {
     app.onprogress = this.handleProgress;
   },
 
+  previewApp: function ai_previewApp(app) {
+    if (!this.isMarketPlaceAppActive()) {
+      return;
+    }
+
+    var marketplace = AppWindowManager.getActiveApp();
+    var manifestURL = app.manifestURL;
+    var appURL = app.origin +
+      (app.manifest.launch_path === '/' ? '/' : app.manifest.launch_path);
+
+    marketplace.childWindowFactory.createPreviewWindow({
+      detail: {
+        url: appURL,
+        origin: app.origin,
+        manifestURL: manifestURL
+      }
+    });
+
+    marketplace.element.addEventListener('_closed', (evt) => {
+      this.clearPreviewApps();
+    });
+
+    var previewAppWindow = marketplace.getTopMostWindow();
+    var previewContent = previewAppWindow.element;
+    var msg = {
+      title: 'Press OPTION to',
+      text: 'Add to Apps'
+    };
+
+    previewContent.addEventListener('_opened', (evt) => {
+      var TYPE = window.InteractiveNotifications.TYPE;
+      window.interactiveNotifications.showNotification(TYPE.TOAST, msg);
+    });
+  },
+
+  getPreviewApps: function() {
+    var previewApps = localStorage.getItem('preview-app') || '';
+    return (previewApps === '') ? [] : previewApps.split(',');
+  },
+
+  addPreviewApp: function(manifestURL) {
+    var previewApps = this.getPreviewApps();
+    if (previewApps.indexOf(manifestURL) === -1) {
+      previewApps.push(manifestURL);
+      localStorage.setItem('preview-app', previewApps.join(','));
+    }
+  },
+
+  removePreviewApp: function(manifestURL) {
+    var previewApps = this.getPreviewApps();
+    var removeIndex = previewApps.indexOf(manifestURL);
+    if (removeIndex !== -1) {
+      previewApps.splice(removeIndex, 1);
+    }
+    localStorage.setItem('preview-app', previewApps.join(','));
+  },
+
+  clearPreviewApps: function() {
```

Comment:
Since this method will uninstall real app instances, no need to rename it. However, the current name `clearPreviewApps` is still not clear to me. Maybe `uninstallPreviewApps` is better?